### PR TITLE
Documentation: math mode use in Mantid docs

### LINKED ISSUE #30704: Documentation: math mode use in Mantid docs
This issue is the result of searching whether there are any other broken Latex math-mode formulae, mentioned in issue #30679 .

### Expected behavior

Latex-based math mode in Sphinx documentation is parsed correctly and consistently throughout the documentation of techniques, algorithms and concepts. 

### Actual behavior

There are some minor issues in multiple places with the incorrect or inconsistent formatting. 

### Platforms affected
all

### PR DESCRIPTION
**Pull request by darominski**
This PR is a result of investigation of Latex and math-mode use in Mantid documentation started by #30679 . The use of roman vs italic characters in math-mode of units and functions has been corrected wherever spotted. Intended use of math in algorithm property descriptions using <math></math> blocks has been replaced with form readable by Sphinx. Spotted issues in references and links have also been corrected.

The one spotted issue that this PR does not address is the observed issue with default parameters for certain dbl list properties that are not broken during creation of restructured text by Sphinx and result in overflowing horizontal dimensions of the document. One of the examples is  [EnggFitPeaks](https://docs.mantidproject.org/nightly/algorithms/EnggFitPeaks-v1.html) algorithm.

List of changes by category is as follows:
1. Corrected algorithm property description:

`GetEi2`, `ConvertToDiffractionMDWorkspace`, `ConvertToMD` 

2. Changed units and functions to roman:

`AnvredCorrection`, `EnggCalibrateFull`, `He3TubeEfficiency`, `IntegrateEllipsoids`, `IntegratePeaksMD`-v2, `LorentzCorrection`, `SpecularReflectionPositionCorrect`, `DiffRotDiscreteCircle`, `EISFDiffCylinder`, `GramCharlier`, `HallRoss`, `InelasticDiffRotDiscreteCircle`, `Meier`, `MultivariateGaussianComptonProfile`, `Voigt`

3. Corrected or improved references to either bibliography or other algorithms:

`D7YIGPositionCalibration`, `DetectorEfficiencyCorUser`, `DivideMD`, `LoadVesuvio`, `NormaliseByDetector`, `RefinePowderInstrumentParameters`, `PolarisedDiffractionILL`

4. Other formatting changes:

`ComputeIncoherentDOS`, `ConvertCWSDExpToMomentum`, `GetEi`-v2, `XrayAbsorptionCorrection`, `FickDiffusion`, `LogNormal`, `GenerateEventsFilter`

**To test:**

1. Build the documentation with docs-html as target
2. Make sure the introduced changes are well formatted and work as described.

Fixes #30704. 

*This does not require release notes* because it is a change of formatting of math mode and corrections of typos in documentation.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/algorithms/IntegrateEllipsoids-v1.rst [hunk 1]
```diff
@@ -187,15 +187,15 @@ For every point on the edge, the trajectory in reciprocal space is a straight li
 Calculate a point at a fixed momentum, say k=1.
 Q in the lab frame:
 
-:math:`\vec{E}=(-k*sin(\theta)*cos(\phi),-k*sin(\theta)*sin(\phi),k-k*cos(\phi))`
+:math:`\vec{E}=(-k \cdot \sin(\theta) * \cos(\phi), -k \cdot \sin(\theta) \cdot \sin(\phi), k - k \cdot \cos(\phi))`
```
Comment (StephenSmith25) [inline docs/source/algorithms/IntegrateEllipsoids-v1.rst:190]: How come this uses two different multiplication symbols?

File: docs/source/fitting/fitfunctions/FickDiffusion.rst [hunk 2]
```diff
@@ -13,7 +13,11 @@ Models the Q dependence of the QENS line width (Gamma (hwhm)), diffusion coeffic
 residence times (tau) and jump lengths (l) to extract the associated long range diffusive
 motions of molecules. Fick's law for diffusion has the form:
 
-.. math:: Gamma(Q) = D*Q^2 where D=<l^2>*Q^2/6*tau
+.. math::
+
+   Gamma(Q) = D \cdot Q^{2},
+
+where :math:`D=<l^{2}> \cdot \frac{Q^2}{6} \cdot tau`
```
Comment (StephenSmith25) [inline docs/source/fitting/fitfunctions/FickDiffusion.rst:20]: I think the spacing with <> doesn't quite look right, maybe it should be 
```suggestion
where :math:`D=\langle{l^2}\rangle \cdot \frac{Q^2}{6} \cdot tau`
```
or prepend \left and \right onto the symbols.

**Review by StephenSmith25 — COMMENTED** [review]
Formatting all looks good, nice work. Just some minor comments.

File: docs/source/algorithms/IntegrateEllipsoids-v1.rst (hunk 1 above)
Comment (darominski) [reply docs/source/algorithms/IntegrateEllipsoids-v1.rst:190]: Well spotted, this is an omission, all multiplication symbols should be created by `\cdot`.

**Review by StephenSmith25 — APPROVED** [review]
Looks good 👍

**Pull request merged**